cpp/velox/memory/VeloxMemoryManager.cc

```diff
@@ -165,6 +162,43 @@ class ListenableArbitrator : public velox::memory::MemoryArbitrator {
   gluten::AllocationListener* listener_;
 };
 
+VeloxMemoryManager::VeloxMemoryManager(
+    std::string name,
+    std::shared_ptr<MemoryAllocator> allocator,
+    std::shared_ptr<AllocationListener> listener)
+    : MemoryManager(), name_(name), listener_(std::move(listener)) {
+  auto veloxAlloc = velox::memory::MemoryAllocator::getInstance();
+  glutenAlloc_ = std::make_shared<ListenableMemoryAllocator>(allocator.get(), listener_);
+#ifdef GLUTEN_ENABLE_HBM
+  auto wrappedAlloc = std::make_shared<VeloxMemoryAllocator>(allocator.get(), veloxAlloc);
+#endif
+  velox::memory::MemoryArbitrator::Config arbitratorConfig{
+      velox::memory::MemoryArbitrator::Kind::kNoOp, // do not use shared arbitrator as it will mess up the thread
+                                                    // contexts (one Spark task reclaims memory from another)
+      velox::memory::kMaxMemory, // the 2nd capacity
+      0,
+      32 << 20,
+      true};
+  velox::memory::IMemoryManager::Options mmOptions{
+      velox::memory::MemoryAllocator::kMaxAlignment,
+      velox::memory::kMaxMemory, // the 1st capacity, Velox requires for a couple of different capacity numbers
+      true,
+      false,
+#ifdef GLUTEN_ENABLE_HBM
+      wrappedAlloc.get(),
+#else
+      veloxAlloc,
+#endif
```

Comment:
Keep VeloxMemoryAllocator and use it when GLUTEN_ENABLE_HBM, and use Velox's allocator by default. @zhztheplayer